content/content.go

```diff
@@ -32,6 +32,13 @@ type ReaderAt interface {
 	Size() int64
 }
 
+// ReaderDigester implements an interface for a Reader that is digesting the data it returns
+type ReaderDigester interface {
```

Comment:
This is an odd interface. We haven't needed it before. There is already a way to write content to a content store and get a digest.

Reply:
Do you need to handle the case where the digest is not known before you start the write? I think this can be handled with a helper in `content`. Let me know if that is not quite what you're looking for.

Reply:
Back then I didn't find anything that would give me the hash of the data that was written via Copy/CopyN so that's why I introduced this interface so we can write the layer descriptor with the size and hash after writing out the encrypted layer.

Reply:
Fixed now with using Writer side hash,